modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/computation/BulkScrollerComputation.java

```diff
@@ -169,21 +176,33 @@ protected void processRecord(ComputationContext context, Record record) {
                         return;
                     }
                     List<String> docIds = scroll.next();
-                    documentIds.addAll(docIds);
+                    int scrollCount = docIds.size();
+                    if (documentCount + scrollCount < queryLimit) {
+                        documentIds.addAll(docIds);
+                    } else {
+                        scrollCount = Math.toIntExact(queryLimit - documentCount);
+                        documentIds.addAll(docIds.subList(0, scrollCount));
+                        limitReached = true;
+                    }
```

Comment:
What about if we get exactly to the queryLimit value? Then we flag limitReached=true but there's no additional documents. The use of this for paging is not clear.

Reply:
the limit should be taken as LIMIT in SQL, the bulk service is not done for paging and we don't have to go into an off-by-one problem here.